### PR TITLE
bump yip and dependants

### PR DESCRIPTION
**Pull request by Itxaka**
New yip version uses  retries for re-reading the partitions so it doesn't fail or skip on the first run.

Also adds sync into the inmutable rootfs

Fixes: https://github.com/rancher-sandbox/cOS-toolkit/issues/887

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: packages/installer/cos.sh [hunk 1]
```diff
@@ -192,7 +192,7 @@ prepare_passive() {
 
 part_probe() {
     local dev=$1
-    partprobe ${dev} 2>/dev/null || true
+    blockdev --rereadpt ${dev} 2>/dev/null || true
```
Comment (Itxaka) [inline packages/installer/cos.sh:195]: This may also fix the issues I was seeing on the tests

**Review by mudler — APPROVED** [review]
Looks good here!

File: packages/installer/cos.sh [hunk 2]
```diff
@@ -405,7 +405,6 @@ get_iso()
 get_image()
 {
     if [ -n "$UPGRADE_IMAGE" ]; then
-        part_probe
```
Comment (Itxaka) [inline packages/installer/cos.sh:408]: @mudler any idea why was this here? This did nothing as there was no `$dev` passed to the call, but didnt fail due to the `|| true` in the function.

Comment (mudler) [reply packages/installer/cos.sh:408]: looks like a leftover, `blkid_probe` was actually probably meant here

File: packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh [hunk 3]
```diff
@@ -29,7 +29,7 @@ install() {
     # Include utilities required for cos-setup services,
     # probably a devoted cos-setup dracut module makes sense
     inst_multiple -o \
-        partprobe lsblk sgdisk mkfs.ext2 mkfs.ext3 mkfs.ext4 mkfs.vfat mkfs.fat mkfs.xfs blkid e2fsck resize2fs mount xfs_growfs umount
+        blockdev lsblk sgdisk mkfs.ext2 mkfs.ext3 mkfs.ext4 mkfs.vfat mkfs.fat mkfs.xfs blkid e2fsck resize2fs mount xfs_growfs umount
```
Comment (mudler) [inline packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh:32]: I just thought at it. But this is the kind of changes that forces the package to depend on a version of yip >= 0.9.22 (in runtime)

Comment (Itxaka) [reply packages/immutable-rootfs/30cos-immutable-rootfs/module-setup.sh:32]: good point! wait for tests to pass then addd it + force-merge? 

Not only the installer but also the inmutablefs package should have a minimum yip dependency as part of this

File: packages/immutable-rootfs/build.yaml [hunk 4]
```diff
@@ -13,7 +13,7 @@ copy:
 - package: 
     category: "toolchain"
     name: "yip"
-    version: ">=0"
+    version: ">=0.9.22"
```
Comment (mudler) [inline packages/immutable-rootfs/build.yaml:16]: This should go in the collection.yaml file of the package as it is a runtime deps, not really a build dep (as we don't run any `yip` command inside the `build.yaml` file

Comment (Itxaka) [reply packages/immutable-rootfs/build.yaml:16]: woops, sorry, screwed it up due checking that raw image test issue, will fix it later after checking that one

Comment (Itxaka) [reply packages/immutable-rootfs/build.yaml:16]: wait, isnt it necessary during build so it can be using during module-setup.sh? thats what it says on the comment no? what would be the difference here on build VS runtime?
